version: "3.7"

services:
  heimdall:
    image: linuxserver/heimdall
    networks:
      - traefik_network
    environment:
      - PGID=100
      - PUID=1024
      - TZ=Europe/Paris
    volumes:
    - /etc/localtime:/etc/localtime:ro
    - config:/config
    deploy:
      labels:
        - "traefik.enable=true"
        - "traefik.http.routers.heimdall-unsecure.rule=Host(`start.{{domain}}`)"
        - "traefik.http.routers.heimdall-unsecure.middlewares=redirect@file"
        - "traefik.http.routers.heimdall-unsecure.entrypoints=http"
        - "traefik.http.routers.heimdall.rule=Host(`start.{{domain}}`)"
        - "traefik.http.routers.heimdall.tls=true"
        - "traefik.http.routers.heimdall.entrypoints=https"
        - "traefik.http.services.heimdall.loadbalancer.server.port=80"
        - "traefik.http.middlewares.auth-heimdall.basicauth.users=admin:$$apr1$$Ux7zvNx5$$ohgAm9jcAtLbnyheCOekZ1"
        - "traefik.http.routers.heimdall.middlewares=auth-heimdall"
        - traefik.docker.network=traefik_public

networks:
  traefik_network:
    external: true

volumes:
  config:
    driver_opts:
      type: "nfs"
      o: "addr={{NAS1}},nolock,soft,rw,timeo=600,vers=4.1"
      device: ":/{{docker_remote_path}}Heimdall/"
